code/CMSProfileController.php

```diff
@@ -2,113 +2,75 @@
+class CMSProfileController extends SingleRecordAdmin
 {
     private static $url_segment = 'myprofile';
 
     private static $menu_title = 'My Profile';
 
     private static $required_permission_codes = 'CMS_ACCESS';
 
-    private static $tree_class = Member::class;
+    private static $model_class = Member::class;
 
     private static $ignore_menuitem = true;
 
-    public function getEditForm($id = null, $fields = null)
-    {
-        $this->setCurrentPageID(Security::getCurrentUser()->ID);
-
-        $form = parent::getEditForm($id, $fields);
-
-        if ($form instanceof HTTPResponse) {
-            return $form;
-        }
-
-        $form->Fields()->removeByName('LastVisited');
-        $form->Fields()->push(new HiddenField('ID', null, Security::getCurrentUser()->ID));
-        $form->Actions()->push(
-            FormAction::create('save', _t(CMSMain::class . '.SAVE', 'Save'))
-                ->addExtraClass('btn-primary font-icon-save')
-                ->setUseButtonTag(true)
-        );
-
-        $form->Actions()->removeByName('action_delete');
+    private static bool $only_one_record = false;
 
-        if ($member = Security::getCurrentUser()) {
-            $form->setValidator($member->getValidator());
-        } else {
-            $form->setValidator(Member::singleton()->getValidator());
-        }
+    private static bool $allow_new_record = false;
 
-        if ($form->Fields()->hasTabSet()) {
-            $form->Fields()->findOrMakeTab('Root')->setTemplate('SilverStripe\\Forms\\CMSTabSet');
+    protected function init()
+    {
+        parent::init();
+        if (!$this->getResponse()->isRedirect()) {
+            $this->setCurrentPageID(Security::getCurrentUser()->ID);
         }
+    }
 
-        $form->addExtraClass('member-profile-form root-form cms-edit-form center fill-height');
-
+    public function getEditForm($id = null, $fields = null): Form
+    {
+        $form = parent::getEditForm($id, $fields);
+        $form->Fields()->push(HiddenField::create('ID', null, Security::getCurrentUser()->ID));
         return $form;
     }
 
     public function canView($member = null)
     {
         $currentUser = Security::getCurrentUser();
-
         if (!$member && $member !== false) {
             $member = $currentUser;
         }
-
-        // cms menus only for logged-in members
+        // cms menus are only for logged-in members
         if (!$member) {
             return false;
         }
-
         // Check they are trying to edit themselves and they have permissions
         return $member->ID === $currentUser->ID && parent::canView($member);
     }
 
     public function save(array $data, Form $form): HTTPResponse
     {
-        $member = Member::get()->byID($data['ID']);
-        if (!$member) {
-            $this->httpError(404);
+        // Make sure the ID is a positive number
+        $id = $data['ID'] ?? null;
+        if (!is_numeric($id) || $id < 1) {
```

Comment:
Should validate that it's an int (or string int), is_numeric() will allow a float

Reply:
Done